openmc/mgxs_library.py

```diff
@@ -1634,6 +1681,285 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
+
+            elif self.scatter_format == 'histogram':
+                # The histogram format does not have enough information to
+                # convert to the other forms without inducing some amount of
+                # error. We will make the assumption that the center of the bin
+                # has the value of the bin. The mu=-1 and 1 points will be
+                # extrapolated from the shape.
+                mu_midpoint = np.linspace(-1, 1, self.num_orders,
+                                          endpoint=False)
+                mu_midpoint += (mu_midpoint[1] - mu_midpoint[0]) * 0.5
+                interp = interp1d(mu_midpoint, orig_data,
+                                  fill_value='extrapolate')
+                # Now get the distribution normalization factor to take from
+                # an integral quantity to a point-wise quantity
+                norm = float(self.num_orders) / 2.0
+
+                # We now have a tabular distribution in tab_data on mu_self.
+                # We now proceed just like the tabular branch above.
+                if target_format == 'legendre':
+                    # find the legendre coefficients via integration. To best
+                    # use the vectorized integration capabilities of scipy,
+                    # this will be done with fixed sample integration routines.
+                    mu_fine = np.linspace(-1, 1, _NMU)
+                    y = [interp(mu_fine) * norm * eval_legendre(l, mu_fine)
+                         for l in range(xsdata.num_orders)]
+                    for l in range(xsdata.num_orders):
+                        new_data[..., l] = simps(y[l], mu_fine)
+
+                    # Remove the very small values resulting from numerical
+                    # precision issues
+                    new_data[..., np.abs(new_data) < 1.E-10] = 0.
+
+                elif target_format == 'tabular':
+                    # Simply use an interpolating function to get the new data
+                    mu = np.linspace(-1, 1, xsdata.num_orders)
+                    new_data[..., :] = interp(mu) * norm
+
+                elif target_format == 'histogram':
+                    # Use an interpolating function to do the bin-wise
+                    # integrals
+                    mu = np.linspace(-1, 1, xsdata.num_orders + 1)
+
+                    # Like the tabular -> legendre path above, this code will
+                    # be written to utilize the vectorized integration
+                    # capabilities instead of having an isotropic and
+                    # angle representation path.
+                    for h_bin in range(xsdata.num_orders):
+                        mu_fine = np.linspace(mu[h_bin], mu[h_bin + 1], _NMU)
+                        new_data[..., h_bin] = \
+                            norm * simps(interp(mu_fine), mu_fine)
+
+                    # Remove the very small values resulting from numerical
```

Comment:
This line of code is repeated four times in different blocks. Could it instead be applied to `new_data` once at the very end, regardless of the self and target representations?